pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
@markturansky does this turn the secret volume into an empty struct?

Also not sure about setSecret, what if it becomes part of the signature of
newBuilder?
On Tue, Apr 21, 2015 at 2:53 PM Mark Turansky <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6578#discussion_r28810228>
> :
>
> > +	EndpointsName string `json:"endpoints"`
> > +	// Required: RBDImage is the rados image name
> > +	RBDImage string `json:"image"`
> > +	// Required: Filesystem type to mount.
> > +	// Must be a filesystem type supported by the host operating system.
> > +	// Ex. "ext4", "xfs", "ntfs"
> > +	// TODO: how do we prevent errors in the filesystem from compromising the machine
> > +	FSType string `json:"fsType,omitempty"`
> > +	// Optional: RBDPool is the rados pool name,default is rbd
> > +	RBDPool string `json:"pool"`
> > +	// Optional: RadosUser is the rados user name, default is admin
> > +	RadosUser string `json:"user"`
> > +	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
> > +	Keyring string `json:"keyring"`
> > +	// Optional: Secret is the authentication secret for RBDUser, default is empty.
> > +	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
>
> I can whip out of POC of this pretty easily. I'll trade for a binder
> review ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6578/files#r28810228>
> .
>